decompiler/pipeline/controlflowanalysis/loop_utility_methods.py

```diff
@@ -210,3 +228,89 @@ def _requirement_without_reinitialization(ast: AbstractSyntaxTree, node: Abstrac
+    return (
+        isinstance(expression, BinaryOperation)
+        and expression.operation in {OperationType.plus, OperationType.minus}
+        and any(isinstance(operand, Constant) for operand in expression.subexpressions())
+        and any(isinstance(operand, Variable) for operand in expression.subexpressions())
+    )
+
+
+def _get_variable_in_binary_operation(binaryoperation: BinaryOperation) -> Variable:
+    """Returns the used variable of a binary operation if available."""
+    for operand in binaryoperation.subexpressions():
+        if isinstance(operand, Variable):
+            return operand
+    return None
+
+
+def _count_unaryoperations_negations(expression: Expression) -> int:
+    """Counts the amount of UnaryOperation negations of an expression."""
+    negations = 0
+    for subexpression in expression.subexpressions():
+        if isinstance(subexpression, UnaryOperation) and subexpression.operation == OperationType.negate:
+            negations += 1
+    return negations
+
+
+def _unify_binary_operation_in_assignment(assignment: Assignment):
+    """Brings a simple binary operation of an assignment into a unified representation like 'var = -var + const' instead of 'var = const - var'."""
+    if not assignment.value.operation == OperationType.plus:
+        assignment.substitute(assignment.value, BinaryOperation(OperationType.plus, [assignment.value.left, assignment.value.right]))
+        assignment.substitute(assignment.value.right, UnaryOperation(OperationType.negate, [assignment.value.right]))
+
+    if any(isinstance(operand, Constant) for operand in assignment.value.left.subexpressions()):
+        assignment.substitute(assignment.value, BinaryOperation(OperationType.plus, [assignment.value.right, assignment.value.left]))
+
+
+def _substract_continuation_from_last_definition(code_node: CodeNode, continuation: AstInstruction, variable_init: AstInstruction):
+    """
+    Substracts the value of the continuation instruction from the last definition, which must be a simple binary operation or a constant,
+    defining the same value as the continuation instruction in the given code node.
+
+    :param code_node: Code node whose last instruction is to be changed
+    :param continuation: Instruction defining the for-loops modification
+    :param variable_init: Instruction defining the for-loops declaration
+    """
+    last_definition = code_node.instructions[_get_last_definition_index_of(code_node, variable_init.instruction.destination)]
+    last_definition.substitute(
+        last_definition.value, BinaryOperation(OperationType.minus, [last_definition.value, continuation.instruction.value.right])
+    )
+    if _count_unaryoperations_negations(continuation.instruction.value.left) % 2:
+        last_definition.substitute(last_definition.value, UnaryOperation(OperationType.negate, [last_definition.value]))
```

Comment:
Again, you define the wrong value. See example.

Reply:
Not sure if I get this correctly. Is it still relevant after the commits?

Reply:
This should be correct now, You could think about only one substitution, but it is okay.